scripts/header_parser.py

```diff
@@ -90,112 +154,113 @@ def __init__(self,
 
 
-def create_wrapper_for_function(fun: CppMethod):
+def parse_function_info(fun: CppMethod) -> tuple[str, list[Parameter]]:
+    """
+    Extracts the name and parameters from a given CppMethod object
+
+    :param fun: A CppMethod extracted by robotpy-cppheaderparser
+    :return:
+    """
+    func_name = fun['name']
+    params_list = []
+    for param_desc, param_doxy, i in zip(
+            fun['parameters'],
+            param_regex.finditer(fun['doxygen']),
+            range(len(fun['parameters']))
+    ):
+        param_mode = param_doxy.group('mode')
+        array_info = array_regex.search(param_doxy.group(0))
+        param_size = array_info.group('lname') if array_info else None
+        param_doc = param_doxy.group('desc')
+        try:
+            params_list.append(Parameter(
+                param_desc, i, param_mode, param_size, param_doc
+            ))
+        except ValueError as e:
+            message = (
+                'Cannot create wrapper for function '
+                '{} (defined in {})'
+            ).format(fun["name"], fun["filename"])
+            raise ValueError(message) from e
+    return func_name, params_list
+
+
+def create_dot_C_wrapper(func_name: str, params_list: list[Parameter]) -> str:
     """
     Uses a function as parsed by robotpy-cppheaderparser together with its
     doxygen comment in order to create an R wrapper
 
     Example: the function defined as follows
-
-    ```
-    /**
-     * @brief description
-     * @param[in] param1
-     * @param[in,out] param2 array of length param1
-     * @param[out] param3 array of length param1
-     */
-    int some_func(const int *param1, int *param2, float *param3);
-    ```
+    ::
+        /**
+         * @brief description
+         * @param[in] param1
+         * @param[in,out] param2 array of length param1
+         * @param[out] param3 array of length param1
+         */
+        int some_func(const int *param1, int *param2, float *param3);
 
     Will produce a wrapper like this
-
-    ```
-    some.func <- function(param2){
-      param1 <- max(length(param2))
-      param2 <- as.integer(param2)
-      param3 <- vector(mode="integer", length=param1)
-      AUTO_RET_PARAMS <- .C("some_func", param1, param2, param3)
-      param2 <- AUTO_RET_PARAMS$param2
-      param3 <- AUTO_RET_PARAMS$param3
-      AUTO_RETVAL <- list("param2" = param2, "param3" = param3
-      return(AUTO_RETVAL)
-    }
-    ```
-
-    :param fun: The function signature together with doxygen comment
-    :type fun: CppMethod
+    ::
+        some.func <- function(param2){
+          param1 <- max(length(param2))
+          param2 <- as.integer(param2)
+          param3 <- vector(mode="integer", length=param1)
+          AUTO_RET_PARAMS <- .C("some_func", param1, param2, param3)
+          param2 <- AUTO_RET_PARAMS$param2
+          param3 <- AUTO_RET_PARAMS$param3
+          AUTO_RETVAL <- list("param2" = param2, "param3" = param3
```

Comment:
Missing `)` ?

Reply:
That was my mistake when making an example, already fixed